src/tracked_struct.rs

```diff
@@ -648,17 +652,15 @@ where
         // that is still live.
 
         {
-            // SAFETY: Guaranteed by caller.
-            let data = unsafe { &*data_raw };
-
-            let last_updated_at = data.updated_at.load();
+            // SAFETY: `updated_at` is never exclusively borrowed, so borrowing it is sound
+            let last_updated_at = unsafe { (*data_raw).updated_at.load() };
```

Comment:
The way the fields are read here (and in a few other places) still creates a temporary `&ValueWithMetadata`. You'd need a raw borrow to avoid that aliasing (or put an `UnsafeCell` around the relevant fields).

Reply:
Really? That is news to me, I thought a field projection like this here will only create a reference to the field?